src/common/app-link/app-link.tsx

```diff
@@ -13,29 +13,35 @@ import type { Params, RouteName } from "../../routes/routes";
   defined in this project. To link outside, use <a></a> tags as usual.
 */
 
+type Route = {
+  pathParams?: Params;
+  queryParams?: Params;
+  routeName: RouteName;
+};
+
 // Extract pathParams from the routeName
-type AppLinkProps<R extends RouteName> = {
+type AppLinkProps = {
```

Comment:
I'm wondering if it makes sense to do something like `interface AppLinkProps extends <whatever the Link props type is>`